src/ui/QuerySummary/QuerySummary.ts

```diff
@@ -189,19 +234,123 @@ export class QuerySummary extends Component {
     }
   }
 
+  private updateNoResultsContainerSnapshot() {
+    const noResultsContainer = this.getNoResultsContainer();
+    if (this.noResultsContainerSnapshot == null && noResultsContainer) {
+      this.noResultsContainerSnapshot = noResultsContainer.innerHTML;
+    }
+  }
+
+  private updateQueryTagsInNoResultsContainer() {
+    const noResultsContainer = this.getNoResultsContainer();
+    if (noResultsContainer) {
+      noResultsContainer.innerHTML = this.noResultsContainerSnapshot;
+      this.parseQueryTagsInContainer(noResultsContainer);
+    }
+  }
+
+  private parseQueryTagsInContainer(node: Node) {
+    let nextNode: Node;
+    if (node.nodeType === Node.ELEMENT_NODE) {
```

Comment:
humm...

Why are we doing all of this ? This seems way too complicated for something that should be much more straightforward.

Lets say we have a div like that for the no results container
```
<div class="coveo-no-results">
    <div> Something </div>
    <a href="somewhere">foo</a>
    <div>${query}</div>
</div>
```

Then, when we instantiate the component we save a string variable like this : 

```
const initialContentOfContainer = noResultsContainer.innerHTML;
```

Then, when you render, the "parsing" (if we can call that) would be a simple string replace : 

```
noResultsContainer.innerHTML = initialContentOfContainer.replace(/\${query}/, escapedQuery)
```

Reply:
I added this logic to make sure that the queryTags inside the attributes of an HtmlElement is not replaced by the querySearched. We discussed the subject and decided that the queryTags should be replaced even if they are in the HtmlElement's attributes so I will remove this logic!